File: Repeated-Substring-Pattern.java
class Solution {
    public boolean repeatedSubstringPattern(String s) {
        int len = s.length();
        
        for (int i = len / 2; i >= 1; i--) {
            if (len % i == 0) {
                String sub = s.substring(0, i);
                int count = len / i;
                StringBuilder result = new StringBuilder();
                
                for (int j = 0; j < count; j++) {
                    result.append(sub);
                }
                
                if (result.toString().equals(s)) {
                    return true;
                }
            }
        }
        return false;
    }
}

File: Repeated-Substring-Pattern.py
class Solution(object):
    def repeatedSubstringPattern(self, s):
        n = len(s)
        for i in range(n // 2, 0, -1):
            if n % i == 0:
                sub = s[:i]
                if sub * (n // i) == s:
                    return True
        return False
